### PR TITLE
browser-debugging-tricks

### LINKED ISSUE #49: [翻译] 67 种浏览器不想让你知道的调试技巧
https://alan.norbauer.com/articles/browser-debugging-tricks

### PR DESCRIPTION
**Pull request by luojiyin1987**
[PR-70](https://github.dev/FreeCodeCamp-Chengdu/Wiki/pull/70) [PR-70](https://codespaces.new/FreeCodeCamp-Chengdu/Wiki/pull/70) [PR-70](https://gitpod.io/?autostart=true#https://github.com/FreeCodeCamp-Chengdu/Wiki/pull/70) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=FreeCodeCamp-Chengdu&utm_campaign=badge_info)

Closed #49

### CODE REVIEW

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 1]
```diff
@@ -6,248 +6,200 @@ translator: ""
...
 
-You can use Chrome’s `monitor` command line method to easily trace all calls to class methods. E.g. given a class `Dog`
-
-```javascript
-class Dog {2  bark(count) {3    /* ... */4  }5}
-```
+您可以使用 Chrome 浏览器的 `monitor` 命令行方法轻松跟踪对类方法的所有调用。例如，给定一个类 `Dog`
 
 ```javascript
 class Dog {2  bark(count) {3    /* ... */4  }5}
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:148]:
```suggestion
class Dog {
  bark(count) {
    /* ... */
  }
}
```
此处是高亮的代码行号乱入

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 2]
```diff
@@ -6,248 +6,200 @@ translator: ""
...
 
-and you’ll get output in the console:
+就会在控制台中得到输出结果：
 
-```
-> function bark called with arguments: 2
-```
-
-```
+```javascript
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:160]:
```suggestion
```text
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 3]
```diff
@@ -6,248 +6,200 @@ translator: ""
...
-```js
-debugger;
-fn(1);
-```
-
+从您的控制台：
 ```js
-debugger;
-fn(1);
+debugger; fn(1);
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:191]:
```suggestion
debugger;
fn(1);
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 4]
```diff
@@ -6,248 +6,200 @@ translator: ""
...
 ```
 
-And then “Step into next function call” to debug the implementation of `fn`.
+然后 `进入下一个函数调用(Step into next function call)`，以调试 `fn` 的实现。
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:194]:
```suggestion
然后“进入下一个函数调用 (Step into next function call)”，以调试 `fn` 的实现。
```
前者并非代码概念

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 5]
```diff
@@ -259,81 +211,66 @@ window.onhashchange = dbg;
...
-Creating a version of `dbg` that pauses execution without breaking navigation is an exercise left up to the reader.
 
-Also, note that this doesn’t handle when code calls `window.location.replace/assign` directly because the page will immediately unload after the assignment, so there is nothing to debug. If you still want to see the source of these redirects (and debug your state at the time of redirect), in Chrome you can `debug` the relevant methods:
+至于如何创建一个能在不中断导航的情况下暂停执行的 `dbg` 版本，读者可自行决定。
 
-```javascript
-debug(window.location.replace);
-debug(window.location.assign);
-```
+另外，请注意，这种方法无法处理直接调用 `window.location.replace/assign` 的情况，因为页面在赋值后会立即销毁，所以没有任何调试内容。如果你仍然想要查看这些重定向的原网址 (并在重定向时调试你的状态)，在 Chrome 中你可以使用 `debug` 来调试相关的方法：
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:217]:
```suggestion
另外，请注意，这种方法无法处理直接调用 `window.location.replace/assign` 的情况，因为页面在赋值后会立即销毁，所以没有任何调试内容。如果你仍然想要查看这些重定向的原网址（并在重定向时调试你的状态），在 Chrome 中你可以使用 `debug` 来调试相关的方法：
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 6]
```diff
@@ -259,81 +211,66 @@ window.onhashchange = dbg;
...
 ```javascript
 debug(window.location.replace);
 debug(window.location.assign);
 ```
 
-## [Debugging Property Reads](#debugging-property-reads)
+## [调试属性读取操作](#debugging-property-reads)
 
-If you have an object and want to know whenever a property is read on it, use an object getter with a `debugger` call. For example, convert `{configOption: true}` to `{get configOption() { debugger; return true; }}` (either in the original source code or using a conditional breakpoint).
+如果你有一个对象，并想知道它的某个属性何时被读取，可以使用一个带有 `debugger` 调用的对象获取器。例如，将 `{configOption: true}` 转换为 `{get configOption() { debugger; return true; }}` (在原始源代码中或使用条件断点)。
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:226]:
```suggestion
如果你有一个对象，并想知道它的某个属性何时被读取，可以使用一个带有 `debugger` 调用的对象获取器。例如，将 `{configOption: true}` 转换为 `{get configOption() { debugger; return true; }}`（在原始源代码中或使用条件断点）。
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 7]
```diff
@@ -259,81 +211,66 @@ window.onhashchange = dbg;
...
-You can copy interesting information out of the browser directly to your clipboard without any string truncation using the `copy()` console API. Some interesting things you might want to copy:
+您可以使用 `copy()` 控制台 API 将浏览器中有趣的信息直接复制到剪贴板，而无需截断任何字符串。您可能想复制一些有趣的内容：
 
--   Snapshot of the current DOM: `copy(document.documentElement.outerHTML)`
--   Metadata about resources (e.g. images): `copy(performance.getEntriesByType("resource"))`
--   A large JSON blob, formatted: `copy(JSON.parse(blob))`
--   A dump of your localStorage: `copy(localStorage)`
--   Etc.
+-   当前 DOM 的快照：`copy(document.documentElement.outerHTML)`
+-   资源的元数据 (如图像)：`copy(performance.getEntriesByType("resource"))`
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:238]:
```suggestion
-   资源的元数据（如图像）：`copy(performance.getEntriesByType("resource"))`
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 8]
```diff
@@ -259,81 +211,66 @@ window.onhashchange = dbg;
...
-The JS console can be helpful when diagnosing problems with your HTML/CSS.
-
-### [Inspect the DOM with JS Disabled](#inspect-the-dom-with-js-disabled)
+JS 控制台有助于诊断 HTML/CSS 的问题。
 
-When in the DOM inspector press ctrl+\\ (Chrome/Windows) to pause JS execution at any time. This allows you to inspect a snapshot of the DOM without worrying about JS mutating the DOM or events (e.g. mouseover) causing the DOM to change from underneath you.
+### [在禁用 JavaScript 的情况下检查 DOM](#inspect-the-dom-with-js-disabled)
 
-### [Inspect an Elusive Element](#inspect-an-elusive-element)
+在 DOM 检查器中按下 `ctrl+\` (Chrome/Windows) 可以随时暂停 JS 的执行。这样您就可以检查 DOM 的快照，而不必担心 JS 会改变 DOM 或事件 (如鼠标悬停) 会导致 DOM 从您脚下发生变化。
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:249]:
```suggestion
在 DOM 检查器中按下 <kbd>ctrl</kbd> + <kbd>\</kbd> (Chrome/Windows) 可以随时暂停 JS 的执行。这样您就可以检查 DOM 的快照，而不必担心 JS 会改变 DOM 或事件（如鼠标悬停）会导致 DOM 从您脚下发生变化。
```

File: _posts/Article/Translation/browser-debugging-tricks.md [hunk 9]
```diff
@@ -259,81 +211,66 @@ window.onhashchange = dbg;
...
+### [检查一个难以捉摸的元素](#inspect-an-elusive-element)
 
-![Elusive element](/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Felusive-element.495b0945.gif&w=1920&q=75)
+假设您要检查一个 DOM 元素，而该元素只有在有条件的情况下才会出现。检查该元素需要将鼠标移动到该元素上，但当您尝试移动时，该元素却消失了：
 
-To inspect the element you can paste this into your console: `setTimeout(function() { debugger; }, 5000);`. This gives you 5 seconds to trigger the UI, and then once the 5 second timer is up, JS execution will pause and nothing will make your element disappear. You are free to move your mouse to the dev tools without losing the element:
+![Elusive element](https://alan.norbauer.com/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Felusive-element.495b0945.gif&w=1920&q=75)
 
-![Elusive element - inspected](/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Felusive-element-inspected.f5f036b4.gif&w=1920&q=75)
+为了检查元素，您可以将以下内容粘贴到控制台中：`setTimeout(function(){debugger; }, 5000);`。这将为您提供 5 秒钟的时间来触发用户界面，一旦 5 秒计时器计时结束，JS 的执行就会暂停，元素也不会消失。您可以自由地将鼠标移到开发工具上，而不会丢失元素：
```
Comment (TechQuery) [inline _posts/Article/Translation/browser-debugging-tricks.md:257]:
```suggestion
为了检查元素，您可以将以下内容粘贴到控制台中：`setTimeout(function () { debugger; }, 5000);`。这将为您提供 5 秒钟的时间来触发用户界面，一旦 5 秒计时器计时结束，JS 的执行就会暂停，元素也不会消失。您可以自由地将鼠标移到开发工具上，而不会丢失元素：
```

**Review by TechQuery — CHANGES_REQUESTED** [review]

File: _posts/Article/Translation/browser-debugging-tricks.md (hunk 8 above)
Comment (luojiyin1987) [reply _posts/Article/Translation/browser-debugging-tricks.md:253]: 要写成  <kbd>ctrl</kbd> + <kbd>\\</kbd>  需要转义。

**Review by TechQuery — APPROVED** [review]

**Pull request merged**